pennylane/_qubit_device.py

```diff
@@ -306,11 +307,17 @@ def marginal_prob(self, prob, wires=None):
         Returns:
             array[float]: array of the resulting marginal probabilities.
         """
-        wires = list(wires or range(self.num_wires))
+        if wires is None:
+            # no need to marginalize
+            return prob
+
         wires = np.hstack(wires)
         inactive_wires = list(set(range(self.num_wires)) - set(wires))
         prob = prob.reshape([2] * self.num_wires)
-        return np.apply_over_axes(np.sum, prob, inactive_wires).flatten()
+
+        basis_states = np.array(list(itertools.product([0, 1], repeat=len(wires))))
```

Comment:
Adding something like ``# Computing permutation indices based on the wires specified``  might be nice here

Reply:
And adding a similar disclaimer in the docstring as well

Reply:
done!